src/main/java/com/github/manolo8/darkbot/backpage/dispatch/DispatchData.java

```diff
@@ -0,0 +1,121 @@
+package com.github.manolo8.darkbot.backpage.dispatch;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class DispatchData {
+    private int permit, gateUnits, slots, maxSlots;
+
+    private final RetrieverBuilder retrieverBuilder = new RetrieverBuilder();
+    private final Map<String, Retriever> retrievers = new LinkedHashMap<>();
+    private final Map<String, InProgress> progressSlots = new LinkedHashMap<>();
+
+    public int getPermit() {
+        return permit;
+    }
+
+    public void setPermit(int permit) {
+        this.permit = permit;
+    }
+
+    public int getGateUnits() {
+        return gateUnits;
+    }
+
+    public void setGateUnits(int gateUnits) {
+        this.gateUnits = gateUnits;
+    }
+
+    public int getSlots() {
+        return slots;
+    }
+
+    public void setSlots(int slots) {
+        this.slots = slots;
+    }
+
+    public int getMaxSlots() {
+        return maxSlots;
+    }
+
+    public void setMaxSlots(int maxSlots) {
+        this.maxSlots = maxSlots;
+    }
+
+    public Map<String, Retriever> getRetrievers() {
+        return retrievers;
+    }
+
+    public void parseRetriever(String str) {
+        retrieverBuilder.build(str);
+        retrieverBuilder.progress(str);
+    }
```

Comment:
```suggestion
    public void parseRow(String str) {
        if (retrieverBuilder.buildRetriever(str)) return;
        if (retrieverBuilder.buildInProgress(str)) return;
    }
```
probably want this to short-circuit if one is successful, and name it something generic if its going to parse everything.

return false if it's not the correct parser, and true if it is the correct one

Reply:
i think the other way around? if one fails dont do the other one

Reply:
true = correct, false = failed

if one is correct you can return, and not do the other